### PR TITLE
handle invalid images

### PR DESCRIPTION
**Pull request by irexyc**
## Motivation

Handle invalid image with invalid url, invalid base64 or truncated data.

related https://github.com/InternLM/lmdeploy/issues/2302

### CODE REVIEW

File: lmdeploy/vl/utils.py [hunk 1]
```diff
@@ -45,27 +53,35 @@ def load_image_from_base64(image: Union[bytes, str]) -> Image.Image:
...
 
-        # Open the image using PIL
-        img = Image.open(BytesIO(response.content))
-    elif image_url.startswith('data:image'):
-        img = load_image_from_base64(image_url.split(',')[1])
-    else:
-        # Load image from local path
-        img = Image.open(image_url)
+        # check image valid
+        img = img.convert('RGB')
```
Comment (lvhan028) [inline lmdeploy/vl/utils.py:79]: Is there a better way to validate the image?

Comment (irexyc) [reply lmdeploy/vl/utils.py:79]: Currently all models will convert the image to RGB, and checking here will not increase the time consumption

**Review by lvhan028 — APPROVED** [review]

**Review by AllentDan — COMMENTED** [review]
Did not pass the following UT:
```python
def test_load_image_base64():
    im = PIL.Image.new(mode='RGB', size=(200, 200))
    encoded = encode_image_base64(im)
    url = f'data:image/jpeg;base64,{encoded}'
    load_image(url)
    import pytest
    with pytest.raises(ValueError):
        load_image(url[:-100])

```

File: lmdeploy/vl/utils.py (hunk 1 above)
Comment (AllentDan) [reply lmdeploy/vl/utils.py:79]: > Is there a better way to validate the image?

I'm not sure if `load()` function satisfied the need. Besides, if `convert('RGB')` is a must for all the model, we may do it here and remove them from `forward` function.

**Comment by irexyc** [comment]
tr

**Review by AllentDan — APPROVED** [review]
LGTM. We may validate the image outside `generate` function in the future.

**Comment by allen-ash** [comment]
If the image provided by the request is invalid, why bother to generate a fake image? Is this gonna return the wrong answer for the request?

**Comment by irexyc** [comment]
@allen-ash 

A reasonable approach is to return an error code if the inputs is invalid. The reason for doing this is that some people have reported that the service will crash if the input is invalid and using a dummy image requires less code changes. We may use error code in the future.

**Pull request merged**